ksql-functional-tests/src/test/java/io/confluent/ksql/test/tools/TestCaseTest.java

```diff
@@ -231,12 +227,13 @@ public void shouldFailForInvalidToplogy() {
     testCase.verifyTopology();
   }
 
-  private TopologyTestDriverContainer getSampleTopologyTestDriverContainer() {
+  private TopologyTestDriverContainer getSampleTopologyTestDriverContainer(final Optional<Topic> sourceTopic) {
     return TopologyTestDriverContainer.of(
         topologyTestDriver,
-        ImmutableList.of(new KsqlTopic("FOO", "foo_kafka", new KsqlJsonSerdeFactory(), false)),
-        new KsqlTopic("BAR", "bar_kafka", new KsqlJsonSerdeFactory(), false),
-        new Pair<>(WindowType.NO_WINDOW, Long.MIN_VALUE)
+        ImmutableList.of(sourceTopic.isPresent()
+    ? sourceTopic.get()
```

Comment:
nit: spacing. Also this feels like a strange use of `Optional`s -- can't we just have `getSampleTopologyTestDriverContainer()` take no arguments and return the default, and separately write
```
final TopologyTestDriverContainer topologyTestDriverContainer = TopologyTestDriverContainer.of(topologyTestDriver, topic);
```
for the single test that uses the non-default?